tests/unit/optimization/StanStochasticMinimizers_unittest.cc

```diff
@@ -0,0 +1,175 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Elfarouk, Wu Lin
+ */
+
+#include <shogun/optimization/StanFirstOrderSAGCostFunction.h>
+#include <gtest/gtest.h>
+#include "StanStochasticMinimizers_unittest.h"
+#include <shogun/optimization/L2Penalty.h>
+#include <shogun/optimization/SGDMinimizer.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/ConstLearningRate.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/optimization/SVRGMinimizer.h>
+#include <shogun/base/Parameter.h>
+#include <shogun/lib/Map.h>
+#include <shogun/optimization/StandardMomentumCorrection.h>
+#include <shogun/optimization/AdaDeltaUpdater.h>
+#include <shogun/optimization/AdamUpdater.h>
+#include <shogun/optimization/NesterovMomentumCorrection.h>
+#include <shogun/optimization/RmsPropUpdater.h>
+#include <shogun/optimization/AdaptMomentumCorrection.h>
+#include <shogun/optimization/L1PenaltyForTG.h>
+#include <shogun/optimization/InverseScalingLearningRate.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/SMIDASMinimizer.h>
+#include <shogun/optimization/PNormMappingFunction.h>
+using namespace shogun;
+using namespace Eigen;
+using namespace stan::math;
+using std::function;
+
+StanVector generateParameters()
+{
+	StanVector w(3, 1);
+	var x1(0), x2(0), x3(0);
+	w(0, 0) = x1;
+	w(1, 0) = x2;
+	w(2, 0) = x3;
+	return w;
+}
+
+function<var(StanVector*, float64_t)>
+cost_for_ith_datapoints(SGMatrix<float64_t>& X, SGMatrix<float64_t>& y)
+{
+	auto f_i = [X, y](StanVector* w, int32_t idx) {
+		var wx_y = (*w)(0, 0) * X(0, idx) + (*w)(1, 0) * X(1, idx) +
+		           (*w)(2, 0) * X(2, idx) - y(0, idx);
+		var res = wx_y * wx_y;
+		res /= 2;
+		return res;
+	};
+	return f_i;
+}
+
+function<var(StanVector*)> get_tot_cost()
+{
+	auto cost = [](StanVector* v) {
+		var total_cost = v->sum();
+		return total_cost;
+	};
+	return cost;
+}
+
+struct RegressionData
+{
+	RegressionData()
+	{
+		init();
+	}
+	SGMatrix<float64_t> y;
+	SGMatrix<float64_t> x;
+	void init();
+};
+
+void RegressionData::init()
+{
+	// The data is referenced from Wu Lin Unit tests
```

Comment:
Would it make sense to have it in a common place?

Reply:
@iglesias Do you mean to put his name in a more obvious place? If so then I've already put his name in the header (copyrights). No plagiarism here :)

Reply:
No. That comment refers to all the lines below, where x and y are initialized.

Reply:
@iglesias Do you mean you'd like the data to be in a more clear place (Like the top) rather than in the middle?